crossvalidation.py

```diff
@@ -47,12 +47,13 @@
 results_file = os.path.join(outdir, 'results.pkl')
 
 # load spatial and genetic data
-spatial_data = pd.read_csv(os.path.join(outdir, params['spatial_data']), index_col=0).rename(
-        columns={"site_name": "name", "long": "x", "lat": "y"}
+spatial_data = pd.read_csv(params['spatial_data']).rename(
+        columns={"name": "name", "long": "x", "lat": "y"}
```

Comment:
Gotcha! I was having trouble getting it to execute, but I think `index_col=0` was the problem, not the path join stuff. Will revert.

Reply:
I think I must be setting up the `crossvalidation.json` for `setup_crossval_grid.py`. 

The only example input on my end in the repo is `data/nebria/crossvalidation.json`

Without the edits here, if I run

```
python setup_crossval_grid.py data/nebria/crossvalidation.json 3 0.1 100 2 1 10
python crossvalidation.py data/nebria/xval_0/xval_params.json
```
I get `FileNotFoundError: [Errno 2] No such file or directory: 'data/nebria/xval_0/stats.csv'`

Reply:
Tried a few more things and read the code more. *If* I'm reading correctly, `outdir` is the path to one of the `xval_*` directories that `setup_crossval_grid.py` makes, but then it tries to look for the data in those paths and can't find them. The change I made here assumed the .json file passed to `setup_crossval_grid.py` has the relative path to the input files. At least when I tried it with these edits I was getting. 

`FileNotFoundError: [Errno 2] No such file or directory: 'data/nebria/xval_0/stats.csv'`

Reply:
Ah-ha, that's true; So, the problem there is elsewhere, including with `setup_crossval_grid.py`. So, it should always be true that if you're in the same directory as the json file, that the paths it refers to exist. So, for instance, `simulation/bias_saddle/crossvalidation.json` has
```
    "spatial_data": "stats.csv",
    "genetic_data": "pairstats.csv",
```
... but however 
```
$ ls simulation/bias_saddle/{stats.csv,pairstats.csv}
ls: cannot access 'simulation/bias_saddle/stats.csv': No such file or directory
ls: cannot access 'simulation/bias_saddle/pairstats.csv': No such file or directory
```
Ah-ha, that's one problem. Notice that we get the error running even `python3 crossvalidation.py simulation/bias_saddle/crossvalidation.json`

Actually I've got the data files in a subdirectory:
```
$ ls simulation/bias_saddle/out_2874405695390_stats
params.json  rep371579.ibd.png  rep371579.pairstats.csv  rep371579.stats.csv
```
and so in fact I should have `simulation/bias_saddle/crossvalidation.json` like
```
    "spatial_data": "out_2874405695390_stats/rep371579.stats.csv",
    "genetic_data": "out_2874405695390_stats/rep371579.pairstats.csv",
```
However, there's still a problem then with `setup_crossval_grid.py` - it ought to modify the paths so that they're still valid, since the new json files they set up are in subdirectories.

I'll fix these things up.